pkg/secret/secret_test.go

```diff
@@ -1 +1,502 @@
 package secret
+
+import (
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestSecretPull(t *testing.T) {
+	testCases := []struct {
+		secretName         string
+		secretNamespace    string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+	}{
+		{
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			expectedErrorText:   "",
+			secretName:         "test",
+			secretNamespace:    "test",
+		},
+		{
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			expectedErrorText:   "secret object test not found in namespace test",
+			secretName:         "test",
+			secretNamespace:    "test",
+
+		},
+		{
+			expectedError:       true,
+			addToRuntimeObjects: true,
+			expectedErrorText:   "secret name cannot be empty",
+			secretName:          "",
+			secretNamespace:    "test",
+		},
+		{
+			expectedError:       true,
+			addToRuntimeObjects: true,
+			expectedErrorText:  "secret namespace cannot be empty",
+			secretName:         "test",
+			secretNamespace:    "",
+		},
+		{
+			expectedError:       true,
+			addToRuntimeObjects: true,
+			expectedErrorText:  "secret type cannot be empty",
+			secretName:         "test",
+			secretNamespace:    "test",
+			secretType:         "", 
+
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, &operatorv1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testCase.secretName,
+					Namespace: testCase.secretNamespace,
+				},
+			})
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		builderResult, err := Pull(testSettings, testCase.secretName, testCase.secretNamespace)
+
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestSecretCreate(t *testing.T) {
```

Comment:
Please add 1 more test case when secret has invalid definition.
Refer to :https://github.com/openshift-kni/eco-goinfra/blob/5966cfaaf0378411c7307befc09a20b1e59e1df2/pkg/metallb/adresspool_test.go#L212